AlpsSDK/ProximityHandler.swift

```diff
@@ -10,224 +10,127 @@ import Foundation
 import CoreLocation
 import Alps
 
-class ProximityHandler {
-    
-    var contextManager: ContextManager!
-    static var immediateTrigger: [String: ProximityEvent] = [:]
-    static var nearTrigger: [String: ProximityEvent] = [:]
-    static var farTrigger: [String: ProximityEvent] = [:]
-    static var unknownTrigger: [String: ProximityEvent] = [:]
-    var immediateBeacons: [String] = []
-    var nearBeacons: [String] = []
-    var farBeacons: [String] = []
-    var unknownBeacons: [String] = []
+protocol ProximityHandlerDelegate : class {
+    func didRangeBeacons(manager: ContextManager, beacons: [CLBeacon], knownBeacons: [IBeaconTriple])
+}
+
+class ProximityHandler: NSObject, ProximityHandlerDelegate {
     var refreshTimer: Int = 60 * 1000 // timer is in milliseconds
+    var beaconsDetected: [String: [IBeaconTriple]] = [:]
+    var beaconsTriggered: [String: [String: ProximityEvent]] = [:]
     
-    init(contextManager: ContextManager) {
-        self.contextManager = contextManager
+    override init() {
+        super.init()
+        beaconsTriggered["unknown"] = [String: ProximityEvent]()
+        beaconsTriggered["immediate"] = [String: ProximityEvent]()
+        beaconsTriggered["far"] = [String: ProximityEvent]()
+        beaconsTriggered["near"] = [String: ProximityEvent]()
```

Comment:
If we don't init it before, I am not sure it will works. 
Could you suggest me something here.

Reply:
As far as I remember `ProximityHandlerDelegate` requires `NSObject`.

Maybe we can bypass that using `@objc` on delegate methods and drop `NSObject` inheritance?

That's just a thought - not sure if it will work though.

Reply:
I can research for @objc, it is still a bit confusing for me, I didn't use objective-c. Will have a look at it.